core/tests/multiplex.rs

```diff
@@ -74,19 +73,14 @@ fn client_to_server_outbound() {
     // A client opens a connection to a server, then an outgoing substream, then sends a message
     // on that substream.
 
-    let (tx, rx) = mpsc::channel();
+    let (tx, rx) = transport::connector();
 
     let bg_thread = thread::spawn(move || {
-        let transport = TcpConfig::new()
+        let future = rx
             .with_upgrade(multiplex::MplexConfig::new())
-            .into_connection_reuse();
-
-        let (listener, addr) = transport
-            .listen_on("/ip4/127.0.0.1/tcp/0".parse().unwrap())
-            .unwrap_or_else(|_| panic!());
-        tx.send(addr).unwrap();
-
-        let future = listener
+            .into_connection_reuse()
+            .listen_on("/memory".parse().unwrap())
+            .unwrap_or_else(|_| panic!()).0
```

Comment:
why not just `unwrap()`?

Reply:
The return value doesn't implement `Debug`. See #367